src/index.ts

```diff
@@ -0,0 +1,482 @@
+		} else {
+			this.restApi = null;
+		}
+		if (config.port) {
+			this.oscApi = new ArenaOscApi(config.host, config.port, this.oscSend.bind(this));
+		} else {
+			this.oscApi = null;
+		}
+		this.setupFeedback();
+		this.setActionDefinitions(this.actions);
+		this.pollStatus();
+	}
+
+	/**
+	 * Poll for in use status continuously until there are
+	 * no more subscriptions or until the module is destroyed
+	 * @return {void}
+	 */
+	public async pollStatus(): Promise<void> {
+		if (this.isPolling) {
+			return;
+		}
+		this.isPolling = true;
+		try {
+			// loop until we don't need to poll any more
+			while (this.restApi) {
+				// check the status via the api
+				try {
+					// only poll status if there are no other subscriptions
+					if (!this.hasPollingSubscriptions) {
+						await this.restApi?.productInfo();
+					}
+					await this.clipUtils.poll();
+					await this.layerUtils.poll();
+					this.updateStatus(InstanceStatus.Ok);
+				} catch (e: any) {
+					this.updateStatus(InstanceStatus.UnknownError, e.message);
+				}
+				await sleep(500);
+			}
+			if (!this.restApi) {
+				// no way to tell if OSC is connected
+				this.updateStatus(InstanceStatus.Ok);
+			}
+		} finally {
+			this.isPolling = false;
+		}
+	}
+
+	private hasPollingSubscriptions(): boolean {
+		return this.clipUtils.hasPollingSubscriptions() || this.layerUtils.hasPollingSubscriptions();
+	}
+
+	/**
+	 * Provide a simple return
+	 * of the necessary fields for the
+	 * instance configuration screen.
+	 * @return {object[]}
+	 */
+	getConfigFields(): SomeCompanionConfigField[] {
+		return configFields();
+	}
+
+	getRestApi(): ArenaRestApi | null {
+		return this.restApi;
+	}
+
+	getOscApi(): ArenaOscApi | null {
+		return this.oscApi;
+	}
+
+	/**
+	 * Clean up the instance before it is destroyed.
+	 * This is called both on shutdown and when an instance
+	 * is disabled or deleted. Destroy any timers and socket
+	 * connections here.
+	 * @return {void}
+	 */
+	async destroy(): Promise<void> {
+		this.restApi = null;
+		this.oscApi = null;
+	}
+}
+
+function getUpgradeScripts(): CompanionStaticUpgradeScript<ResolumeArenaConfig>[] {
+	return [
+		function (
+			_context: CompanionUpgradeContext<ResolumeArenaConfig>,
+			props: CompanionStaticUpgradeProps<ResolumeArenaConfig>
+		): CompanionStaticUpgradeResult<ResolumeArenaConfig> {
+			// upgrade_v1_0_4
+			let updateActions = [];
+
+			for (const action of props.actions) {
+				switch (action.actionId) {
+					case 'custom':
+						if (action.options !== undefined && action.options.customCmd !== undefined) {
+							action.options.customPath = action.options.customCmd;
+							delete action.options.customCmd;
+							updateActions.push(action);
+						}
+						break;
+				}
+			}
+
+			return {
+				updatedConfig: null,
+				updatedActions: updateActions,
+				updatedFeedbacks: [],
+			};
+		},
+	];
+}
+
+exports = module.exports = ResolumeArenaModuleInstance;
```

Comment:
This line isnt needed, but doesnt do any harm

Reply:
gone